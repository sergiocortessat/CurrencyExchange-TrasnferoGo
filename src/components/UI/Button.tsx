import React from 'react';

interface ButtonProps {
  variant: 'contained' | 'outlined' | 'text';
  color: 'primary' | 'secondary';
  onClick: () => void;
  children: React.ReactNode;
  disabled: boolean;
  className?: string;
}

const Button: React.FC<ButtonProps> = (props) => {
  const { variant, color, onClick, disabled, children, className = "btn" } = props;
  return (
    <button
      className={`custom-button ${variant} ${color}-color ${className} custom-button--${!disabled}`}
      onClick={onClick}
      disabled={!disabled}
    >
      {children}
    </button>
  );
};

export default Button;
