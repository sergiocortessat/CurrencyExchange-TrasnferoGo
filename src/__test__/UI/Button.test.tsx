import React from 'react';
import { render, screen } from '@testing-library/react';
import Button from '../../components/UI/Button';

describe('Button', () => {
  it('should render with the correct props', () => {
    const onChange = jest.fn();
    const { asFragment } = render(
      <Button variant="contained" color="primary" onClick={onChange} disabled={false}>
        Click Me
      </Button>
    );

    const buttonElement = screen.getByText('Click Me');

    expect(asFragment()).toMatchSnapshot();
    expect(buttonElement).toHaveClass('custom-button');
    expect(buttonElement).toHaveClass('primary-color');
    expect(buttonElement).not.toHaveClass('outlined');
    expect(buttonElement).not.toHaveClass('text');
    expect(buttonElement).not.toHaveClass('secondary-color');
    expect(buttonElement).toHaveClass('btn');
    expect(buttonElement).toHaveAttribute('disabled');
  });

  it('should render with disabled state', () => {
    const { asFragment } = render(
      <Button variant="contained" color="primary" onClick={() => {}} disabled={true}>
        Click Me
      </Button>
    );

    expect(asFragment()).toMatchSnapshot();
    const buttonElement = screen.getByText('Click Me');

    expect(buttonElement).not.toHaveAttribute('disabled');
    expect(buttonElement).toHaveClass('custom-button--false');
    expect(buttonElement).not.toHaveClass('custom-button--true');
  });
});
