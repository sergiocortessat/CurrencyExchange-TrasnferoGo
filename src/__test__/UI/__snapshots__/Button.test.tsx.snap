// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render with disabled state 1`] = `
<DocumentFragment>
  <button
    class="custom-button contained primary-color btn custom-button--false"
  >
    Click Me
  </button>
</DocumentFragment>
`;

exports[`Button should render with the correct props 1`] = `
<DocumentFragment>
  <button
    class="custom-button contained primary-color btn custom-button--true"
    disabled=""
  >
    Click Me
  </button>
</DocumentFragment>
`;
